### PR TITLE
remove option to deploy to obsolete runtimes

### LINKED ISSUE #3550: Remove "Allow deploying to obsolete App Engine runtimes"
Per @ludoch I don't think this is possible any more, for anyone. Can we take this checkbox out of the deploy dialog?

image

### PR DESCRIPTION
**Pull request by elharo**
fixes #3550 @chanseokoh @briandealwis

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java [hunk 1]
```diff
@@ -67,38 +64,8 @@ private boolean checkAppEngineRuntimeCompatibility(Shell shell, IProject project
...
-            deployPreferences.setAllowObsoleteRuntime(true);
-            deployPreferences.save();
-          } catch (BackingStoreException ex) {
-            logger.log(
-                Level.WARNING, "Unable to check project use of obsolete App Engine runtime", ex);
-          }
-        }
-        return dialog.getReturnCode() == IDialogConstants.OK_ID;
+      if (AppEngineStandardFacet.usesObsoleteRuntime(facetedProject)) {
+        return false;
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java]: I see one problem that when we identified an obsolete runtime, the deploy operation abruptly halts without any message to the user. (`checkProject()` in `execute()` simply returns null.) The user may believe that the deploy command isn't just working without knowing what the problem is. Somehow, we should let the user know.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java]: Good catch. So maybe I should put the dialog back here but change the message?

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java]: I think that makes sense. The dialog style should change to have a single OK button.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java]: done

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java [hunk 2]
```diff
@@ -67,38 +64,19 @@ private boolean checkAppEngineRuntimeCompatibility(Shell shell, IProject project
...
+        String message = Messages.getString("obsolete.runtime.message", project.getName());
+        MessageDialog dialog = new MessageDialog(
+            shell,
+            Messages.getString("obsolete.runtime.title"),
+            null,
+            message,
+            MessageDialog.ERROR,
+            buttons,
+            0 /* index into buttonsWithIds of default button */);
+        dialog.open();
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java]: I can confirm `MessageDialog.openError(shell, Messages.getString("obsolete.runtime.title"), message);` does the same.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/messages.properties [hunk 3]
```diff
@@ -101,3 +93,8 @@ action.stop=Stop
 action.remove=Remove
 job.terminated.template=<terminated> {0}
 deploy.console.name=App Engine Deploy
+
+ok.button=OK
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/messages.properties]: Unnecessary if you use `MessageDialog.openError`.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/messages.properties]: good, done

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java (hunk 2 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java]: done

**Pull request merged**